### PR TITLE
[PoC] migrate docs workflow to GHA

### PR DESCRIPTION
**Pull request by pmeier**
I don't intend to merge this PR at this point, since we need to resolve the doc preview issue on GHA first. In pytorch/test-infra#3842 @DanilBaibak enabled Dr. CI, i.e. the bot that will comment below, for `torchvision` again. Although it doesn't work yet, the artifact generated by the workflow added in this PR needs to be uploaded to a S3 bucket similar to what is done in [PyTorch core](https://github.com/pytorch/pytorch/blob/92fbb35bffd49a882aca398b831d99e2b304c004/.github/workflows/_docs.yml#L152-L160). If I'm not mistaken, we only need to have a second job that depends on the one I added, that downloads the artifact and uploads it again to S3.

I ignored some intricacies here, like nightly doc upload and release branch version fix:

https://github.com/pytorch/vision/blob/82cf540b45fae6bc83cfff10db9041c3a1aeec5d/.circleci/config.yml#L893-L895

cc @seemethere

### CODE REVIEW

**Comment by pmeier** [comment]
Jobs is green: https://github.com/pytorch/vision/actions/runs/4383643063/jobs/7674140541. Documentation is available as artifact: https://github.com/pytorch/vision/actions/runs/4383643063. I've verified that it is ok.

**Comment by malfet** [comment]
So, upload fails because you need to run the build on pytorch org runner rather than on a stock github one

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        sed -i -e 's/-j auto/-j 1/' Makefile
+        make html
+        echo '::endgroup::'
+        
+        mv build/html "${RUNNER_ARTIFACT_DIR}"
+
+  upload-preview:
+    if: github.event_name == 'pull_request'
+    needs: [build]
+    runs-on: ubuntu-latest
```
Comment (malfet) [inline .github/workflows/docs.yml:55]:
```suggestion
    runs-on: [self-hosted, linux.2xlarge]
```

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      - uses: actions/download-artifact@v3
+        with:
+          name: docs
+
+      - name: Debug
+        run: |
+          sudo apt -y update && sudo apt -y install tree
+          ls -la
+          tree html
+
```
Comment (malfet) [inline .github/workflows/docs.yml:66]: Remove debug step
```suggestion
```

**Comment by malfet** [comment]
I can see doc preview now (need to fix URL in the hud though): https://docs-preview.pytorch.org/vision/7407/index.html

**Pull request merged**